src/java/htsjdk/samtools/sra/SRALazyRecord.java

```diff
@@ -0,0 +1,941 @@
+        if (!initializedFlags.contains(LazyFlag.READ_PAIRED)) {
+            initializedFlags.add(LazyFlag.READ_PAIRED);
+        }
+        super.setReadPairedFlag(flag);
+    }
+
+    @Override
+    public boolean getNotPrimaryAlignmentFlag() {
+        if (!initializedFlags.contains(LazyFlag.NOT_PRIMARY_ALIGNMENT)) {
+            setNotPrimaryAlignmentFlag(getNotPrimaryAlignmentFlagImpl());
+        }
+        return super.getNotPrimaryAlignmentFlag();
+    }
+
+    @Override
+    public void setNotPrimaryAlignmentFlag(final boolean flag) {
+        if (!initializedFlags.contains(LazyFlag.NOT_PRIMARY_ALIGNMENT)) {
+            initializedFlags.add(LazyFlag.NOT_PRIMARY_ALIGNMENT);
+        }
+        super.setNotPrimaryAlignmentFlag(flag);
+    }
+
+    @Override
+    public boolean getMateNegativeStrandFlag() {
+        if (!initializedFlags.contains(LazyFlag.MATE_NEGATIVE_STRAND)) {
+            setMateNegativeStrandFlag(getMateNegativeStrandFlagImpl());
+        }
+        return super.getMateNegativeStrandFlag();
+    }
+
+    @Override
+    public void setMateNegativeStrandFlag(final boolean flag) {
+        if (!initializedFlags.contains(LazyFlag.MATE_NEGATIVE_STRAND)) {
+            initializedFlags.add(LazyFlag.MATE_NEGATIVE_STRAND);
+        }
+        super.setMateNegativeStrandFlag(flag);
+    }
+
+    @Override
+    public boolean getMateUnmappedFlag() {
+        if (!initializedFlags.contains(LazyFlag.MATE_UNMAPPED)) {
+            setMateUnmappedFlag(getMateUnmappedFlagImpl());
+        }
+        return super.getMateUnmappedFlag();
+    }
+
+    @Override
+    public void setMateUnmappedFlag(final boolean flag) {
+        if (!initializedFlags.contains(LazyFlag.MATE_UNMAPPED)) {
+            initializedFlags.add(LazyFlag.MATE_UNMAPPED);
+        }
+        super.setMateUnmappedFlag(flag);
+    }
+
+    @Override
+    public boolean getFirstOfPairFlag() {
+        if (!initializedFlags.contains(LazyFlag.FIRST_OF_PAIR)) {
+            setFirstOfPairFlag(getFirstOfPairFlagImpl());
+        }
+        return super.getFirstOfPairFlag();
+    }
+
+    @Override
+    public void setFirstOfPairFlag(final boolean flag) {
+        if (!initializedFlags.contains(LazyFlag.FIRST_OF_PAIR)) {
+            initializedFlags.add(LazyFlag.FIRST_OF_PAIR);
+        }
+        super.setFirstOfPairFlag(flag);
+    }
+
+    @Override
+    public boolean getSecondOfPairFlag() {
+        if (!initializedFlags.contains(LazyFlag.SECOND_OF_PAIR)) {
+            setSecondOfPairFlag(getSecondOfPairFlagImpl());
+        }
+        return super.getSecondOfPairFlag();
+    }
+
+    @Override
+    public void setSecondOfPairFlag(final boolean flag) {
+        if (!initializedFlags.contains(LazyFlag.SECOND_OF_PAIR)) {
+            initializedFlags.add(LazyFlag.SECOND_OF_PAIR);
+        }
+        super.setSecondOfPairFlag(flag);
+    }
+
+
+    // ===== attributes =====
+
+    @Override
+    public Object getAttribute(final short tag) {
```

Comment:
Nothing bad: it will go to SAMRecord.getAttribute and look up there.
Will check what other attributes SRA have support for